applications/nrf_desktop/doc/hid_reportq.rst

```diff
@@ -0,0 +1,81 @@
+.. _nrf_desktop_hid_reportq:
+
+HID report queue utility
+########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+A HID subscriber (for example, a USB HID class instance handled by :ref:`nrf_desktop_usb_state`) limits the maximum number of HID input reports that can be handled simultaneously.
+Enqueuing reports locally at the source may be necessary to prevent HID report drops.
+The HID report queue utility can be used by an application module to simplify enqueuing HID input reports received from connected HID peripherals before providing them to HID subscriber.
+
+Configuration
+*************
+
+Make sure that heap size (:kconfig:option:`CONFIG_HEAP_MEM_POOL_SIZE`) is large enough.
+Data structures used to enqueue HID reports are dynamically allocated using the :c:func:`k_malloc` function.
+When no longer needed, the structures are freed using the :c:func:`k_free` function.
+
+Use the :ref:`CONFIG_DESKTOP_HID_REPORTQ <config_desktop_app_options>` Kconfig option to enable the utility.
+You can use the utility only on HID dongles (:ref:`CONFIG_DESKTOP_ROLE_HID_DONGLE <config_desktop_app_options>`).
+
+You can also configure the following properties:
+
+* Maximum number of enqueued HID reports (:ref:`CONFIG_DESKTOP_HID_REPORTQ_MAX_ENQUEUED_REPORTS <config_desktop_app_options>`)
+* Number of supported HID report queues (:ref:`CONFIG_DESKTOP_HID_REPORTQ_QUEUE_COUNT <config_desktop_app_options>`)
+
+See Kconfig help for more details.
+
+Using HID report queue
+**********************
+
+The utility acts as a middleware between a connected HID peripheral and a HID subscriber.
```

Comment:
I find this statement a bit confusing - doesn't nrf_desktop_hid_forward act as such middleware?

Reply:
Provided a bit more explanation